csharp/source/autowrap/csharp/boilerplate.d

```diff
@@ -102,11 +104,45 @@ private string commonBoilerplate() @safe pure {
             }
         }
 
+        mixin(generateSharedTypes());
+
         public void pinPointer(void* ptr) nothrow {
             GC.setAttr(ptr, GC.BlkAttr.NO_MOVE);
             GC.addRoot(ptr);
         }
 
+        public datetime toDatetime(T)(T value)
+        if (isDateTimeType!T) {
+            return datetime(value);
+        }
+
+        public datetime[] toDatetime1DArray(T)(T[] value)
```

Comment:
Unless there are 2D arrays, wouldn't `toDatetimes` be a better name?

Reply:
Indeed, however, IIRC according to @John-Colvin there are 2D arrays. The problem right now is that I haven't gotten around to supporting multi-dimensional arrays because Laeeth told me to work on other things first.